omod/src/test/java/org/openmrs/module/idgen/web/controller/IdentifierSourceRestControllerTest.java

```diff
@@ -0,0 +1,132 @@
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertFalse;
+
+public class IdentifierSourceRestControllerTest extends MainResourceControllerTest {
+	
+	public static final String IDENTIFIER_SOURCE_UUID = "0d47284f-9e9b-4a81-a88b-8bb42bc0a999";
+	
+	private IdentifierSourceService service;
+
+	@Before
+	public void setUp() throws Exception {
+		executeDataSet("org/openmrs/module/idgen/include/TestDataForLoadTest.xml");
+		this.service = Context.getService(IdentifierSourceService.class);
+	}
+
+	@Override
+	public String getURI() {
+		return "idgen/identifiersource";
+	}
+
+	@Override
+	public String getUuid() {
+		return IDENTIFIER_SOURCE_UUID;
+	}
+
+	@Override
+	public long getAllCount()  {
+		return service.getAllIdentifierSources(false).size();
+	}
+	
+	@Test
+	public void shouldEditAnIdentifierSource() throws Exception {
+		
+		String description = "{\"description\":\"Updated description\"}";
+		
+		assertNull(service.getIdentifierSourceByUuid(getUuid()).getDescription());
+		
+		handle(newPostRequest(getURI() + "/" + getUuid(), description));
+		
+		assertEquals("Updated description", service.getIdentifierSourceByUuid(getUuid()).getDescription());
+		
+	}
+	
+	@Test
+	public void shouldVoidAnIdentifierSource() throws Exception {
+		
+		final String reason = "none";
+		
+		assertFalse(service.getIdentifierSourceByUuid(getUuid()).isRetired());
+		
+		MockHttpServletRequest req = request(RequestMethod.DELETE, getURI() + "/" + getUuid());
+		req.addParameter("reason", reason);
+		handle(req);
+		
+		assertTrue(service.getIdentifierSourceByUuid(getUuid()).isRetired());
+		assertEquals(reason, service.getIdentifierSourceByUuid(getUuid()).getRetireReason());
+
+	}
+	
+	@Test
+	public void shouldPurgeAnIdentifierSource() throws Exception {
+		
+		assertNotNull(service.getIdentifierSourceByUuid(getUuid()));
+		
+		handle(newDeleteRequest(getURI() + "/" + getUuid(), new Parameter("purge", "")));
+		
+		assertNull(service.getIdentifierSourceByUuid(getUuid()));
+		
+	}	
+
+	@Override
+	@Test(expected = ResourceDoesNotSupportOperationException.class)
+	public void shouldGetAll() throws Exception {
+	    super.shouldGetAll();
+	}
+
+	@Test
+	public void shouldGetAnIdentifierSourceByUuid() throws Exception {
+		
+		SimpleObject result = deserialize(handle(newGetRequest(getURI() + "/" + getUuid())));
+				
+		assertNotNull(result);
+		assertEquals(getUuid(), PropertyUtils.getProperty(result, "uuid"));
+	
+	}
+
+	@Test
+	public void shouldSaveIdentifierSource() throws Exception {
+		
+		long initialIdentifierSourceCount = getAllCount();
+		
+		SimpleObject initialIdentifierSource = new SimpleObject();
+		initialIdentifierSource.add("name", "test identifier source name");
+		initialIdentifierSource.add("description", "test identifier source desc");
+		initialIdentifierSource.add("identifierType", 3);
```

Comment:
We do not use database autogenerated integer identifiers for REST calls. We instead use uuid. So can you replace the above 2?

Reply:
fixed